src/Tasks/Microsoft.NET.Build.Tasks/ResolvePackageDependencies.cs

```diff
@@ -279,8 +279,26 @@ private void RaiseLockFileTargets()
 
         private void GetPackageAndFileDependencies(LockFileTarget target)
         {
-            var resolvedPackageVersions = target.Libraries
-                .ToDictionary(pkg => pkg.Name, pkg => pkg.Version.ToNormalizedString(), StringComparer.OrdinalIgnoreCase);
+            // In case of there are duplicated packages in libraries with differing versions,
+            // We skip package altogether
+            var resolvedPackageVersions = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            var duplicatedPackages = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+            foreach (LockFileTargetLibrary library in target.Libraries)
+            {
+                string libraryName = library.Name;
+                if (duplicatedPackages.Contains(libraryName))
+                {
+                    continue;
+                }
+
+                if (resolvedPackageVersions.ContainsKey(libraryName))
+                {
+                    duplicatedPackages.Add(libraryName);
+                    resolvedPackageVersions.Remove(libraryName);
+                }
+
+                resolvedPackageVersions[libraryName] = library.Version.ToNormalizedString();
```

Comment:
If we let first one pass-through then we can avoid hashset lookup & dict.Remove calls.
If we want to minimize the impact then perhaps try/catch could be useful so we only run additional code only in error case.